(#282) Allow building with the latest SSMS installed.

corbob opened the pull request:
## Description Of Changes

Update the logic to find an MSBuild that actually has an MSBuild?

## Motivation and Context

Being able to build things while the latest SSMS is installed.

## Testing

For any project using this recipe, apply a similar diff to this:

```
diff --git a/recipe.cake b/recipe.cake
index bc029d1f..30269b28 100644
--- a/recipe.cake
+++ b/recipe.cake
@@ -1,4 +1,5 @@
-#load nuget:?package=Chocolatey.Cake.Recipe&version=0.30.1
+#load nuget:?package=Chocolatey.Cake.Recipe&version=0.30.1&Include=/**/version.cake
+#load local:?path=C:/code/chocolatey.cake.recipe/chocolatey.cake.recipe/Content/*.cake
 #tool nuget:?package=WiX&version=3.11.2
 ```

Run the build script.

### Operating Systems Testing

Windows 11

## Change Types Made

* [x] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [ ] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v3 compatibility checked?

## Related Issue

- Fixes #282

corbob committed:
(#282) Find MSBuild tool with an MSBuild.exe

This commit updates the logic for detecting the MSBuild tool. Our builds
encountered this issue: https://github.com/NuGet/Home/issues/14349.

While investigating it, it was found that dnnsoftware encounted a
similar issue and fixed it here:
https://github.com/dnnsoftware/Dnn.Platform/pull/6575. This commit
updates the methods we use to call VSWhere to identify the MSBuild to
return based on actually having an MSBuild.exe file.

gep13 commented:
@corbob our futureselves are going to want/need more words in the commit message than that! 😄

corbob commented:
> @corbob our futureselves are going to want/need more words in the commit message than that! 😄

@gep13 I know that 🤪 Current me was just getting things up while I was waiting for a packer build to finish! But tomorrow me is a future me that will likely want more information, so thank you very much for the nudge today while I have some of the information in my head 👍

corbob committed:
(docs) Update Local Development docs

I've found a better way to do local development on the recipe, so this
updates the Readme with that information.

corbob commented:
This is draft while I work through any details needed here... It's also possible that this is no longer an issue once Visual Studio 2026 is installed.
